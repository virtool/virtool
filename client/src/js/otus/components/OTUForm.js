import React from "react";
import { Row, Col, Modal } from "react-bootstrap";
import { InputError, SaveButton } from "../../base";

const OTUForm = (props) => (
    <form onSubmit={props.handleSubmit}>
        <Modal.Body>
            <Row>
                <Col md={8}>
                    <InputError
                        label="Name"
                        name="name"
                        value={props.name}
                        onChange={props.handleChange}
                        error={props.errorName}
                    />
                </Col>
                <Col md={4}>
                    <InputError
                        label="Abbreviation"
                        name="abbreviation"
                        value={props.abbreviation}
                        onChange={props.handleChange}
                        error={props.errorAbbreviation}
                    />
                </Col>
            </Row>
        </Modal.Body>
        <Modal.Footer>
            <SaveButton pullRight />
        </Modal.Footer>
    </form>
);

export default OTUForm;
