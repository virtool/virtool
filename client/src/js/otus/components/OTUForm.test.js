import OTUForm from "./OTUForm";

describe("<OTUForm />", () => {
    let props;
    let wrapper;

    it("renders correctly", () => {
        props = {
            handleSubmit: jest.fn(),
            handleChange: jest.fn(),
            name: "test",
            abbreviation: "T",
            errorName: "",
            errorAbbreviation: ""
        };
        wrapper = shallow(<OTUForm {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

});
