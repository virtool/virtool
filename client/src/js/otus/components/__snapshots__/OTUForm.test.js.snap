// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OTUForm /> renders correctly 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={8}
      >
        <InputError
          error=""
          label="Name"
          name="name"
          onChange={[MockFunction]}
          value="test"
        />
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        md={4}
      >
        <InputError
          error=""
          label="Abbreviation"
          name="abbreviation"
          onChange={[MockFunction]}
          value="T"
        />
      </Col>
    </Row>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <SaveButton
      pullRight={true}
    />
  </ModalFooter>
</form>
`;
